.github/workflows/pr_genfiles.yml

```diff
@@ -24,21 +24,20 @@ jobs:
       options: --cpus 7
     steps:
     - uses: actions/checkout@v3
-    - name: Go cache
+    - name: go cache
       uses: actions/cache@v3
       with:
         path: /px/pkg/mod
-        key: build-cache-${{hashFiles('go.sum')}}
-    - name: Yarn cache
+        key: go-cache-${{ hashFiles('go.sum') }}
+        restore-keys: |
+          go-cache-
+    - name: yarn cache
       uses: actions/cache@v3
       with:
-        path: ./src/ui/.yarn
-        key: build-cache-${{hashFiles('src/ui/yarn.lock')}}
-    - name: Bazel cache
-      uses: actions/cache@v3
-      with:
-        path: ~/.cache/bazel
-        key: genfile-checker-${{hashFiles('WORKSPACE')}}
+        path: ./src/ui/.yarn/cache
+        key: yarn-cache-${{ hashFiles('src/ui/yarn.lock', 'src/ui/.yarnrc.yml') }}
+        restore-keys: |
```

Comment:
So restore-keys knows to pull from main cache somehow ?

Reply:
Restore keys just lets you use a "stale cache" if the hash doesn't map. So basically if you change `yarn.lock` or `.yarnrc.yml`, you will get a cache miss but then this will populate the yarn cache with the LRU yarn cache and then create a new cache using that as a base.

Pulling from main cache is a actions/cache behavior that Github describes in the doc I linked to.